File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[]{0,0};
        
        while(!s.equals("1")){
            
            StringBuilder sb = new StringBuilder();
            
            for(char ch: s.toCharArray()){
                if(ch == '1'){
                    sb.append("1");
                }else{
                    answer[1]++;
                }
            }
            int length = sb.length();
            
            StringBuilder str = new StringBuilder();
            
            while(length>0){
                str.append(length % 2  );
                length /= 2;
            }
            answer[0]++;
            s = str.reverse().toString();
        }
    
        return answer;
        
    
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    var answer = [];
    
    var zeroCount=0;
    var transCount=0;
    
    
    
    while(parseInt(s) !== 1){
        
        var oneCount=0;
        var arr = s.split('')
        arr.forEach(elem => {
            if(elem==1) oneCount++;
        })
        
        zeroCount+=s.length-oneCount;
        transCount++;
        
        s = oneCount.toString(2);
        
    }
    
    answer[0] = transCount;
    answer[1] = zeroCount;
    
    
    
    
    
    
    
    return answer;
}